### PR TITLE
enabling R.scan to be used as transducer.

### LINKED ISSUE #2816: enabling R.scan to be used as transducer.
Hi!

I noticed that `R.scan` doesn't work as transducer and I think could be useful for performing reducer operations lazily.

### PR DESCRIPTION
**Pull request by xgbuils**
It fixes #2816

### CODE REVIEW

**Comment by CrossEye** [comment]
:herb:

I'm happy with this (as much as I am with anything extending our transducers.)

Anyone else?  @ramda/core?

**Comment by xgbuils** [comment]
Cool, @CrossEye ! 

After self-reviewing my PR I'm going to remove `['scan']` in dispatchable because I think it's out of the scope of this PR.

**Comment by CrossEye** [comment]
As with many things around this time, this PR was inadvertently dropped.  This makes sense, and I wonder, @xgbuils, if you want to resolve the conflicts so that we can merge?

File: source/scan.js [hunk 1]
```diff
@@ -22,7 +24,7 @@ import _curry3 from './internal/_curry3.js';
  *      const factorials = R.scan(R.multiply, 1, numbers); //=> [1, 1, 2, 6, 24]
  * @symb R.scan(f, a, [b, c]) = [a, f(a, b), f(f(a, b), c)]
  */
-var scan = _curry3(function scan(fn, acc, list) {
+var scan = _curry3(_dispatchable([], _xscan, function scan(fn, acc, list) {
```
Comment (customcommander) [inline source/scan.js:27]: If I understand this correctly then `R.scan` can act as a transducer when given a transformer as a list. Perhaps we should mention that in the description?

**Review by customcommander — CHANGES_REQUESTED** [review]

File: source/scan.js (hunk 1 above)
Comment (xgbuils) [reply source/scan.js:29]: Thanks for the observation, @customcommander. I've added that in the description.

**Review by customcommander — DISMISSED** [review]
Approving on the basis that this pull request has been approved before but since there's been some conflicts I think it would make sense for @CrossEye to have a quick re-review.

Thank you @xgbuils and sorry for the delay ;)

**Comment by xgbuils** [comment]
No problem. My pleasure. I have to review documentation about transducers again (I had to review my PR accurately to understand what I did 2 years ago). I think that there are another functions that works with lists that can be adapted to work with transducers and I could review it in the future.

**Review by CrossEye — DISMISSED** [review]

**Review by customcommander — DISMISSED** [review]

**Comment by CrossEye** [comment]
Hmm, just coming back to this to see there are conflicts.  Going to bed now, but I hope to look at these soon.

**Pull request merged**